### PR TITLE
Added Grouping for Arguments

### PR DESCRIPTION
**Pull request by Daddelhai**
Arguments can be grouped. Groups have a Header and optional a description. Every argument is in the default Group "arguments", if no other group is assigned.

Fixes Issue #22 

Example:

```
nitro::broken_options::parser parser("app_name", "about");

parser.group("group1").option("opt", "some opt").short_name("o");
parser.group("group2").toggle("tog", "some toggle").short_name("t");
parser.group("group2").toggle("togg", "some other toggle").short_name("u");
```

Output:

```
usage: app_name --opt

about


group1:
  -o [ --opt ] arg                        some opt

group2:
  -t [ --tog ]                            some toggle
  -u [ --togg ]                           some other toggle
```

### CODE REVIEW

**Comment by bmario** [comment]
@cvonelm Anything missing from the lo2s use case?

File: include/nitro/broken_options/option/group.hpp [hunk 1]
```diff
@@ -38,10 +38,10 @@ namespace nitro
 {
 namespace broken_options
 {
-    class group
+    class option_group
```
Comment (bmario) [inline include/nitro/broken_options/option/group.hpp:41]: That's a misnomer in nitro language. It represents a group of arguments, not options. And while you're on it, rename the options_ member as well.

File: include/nitro/broken_options/parser.ipp [hunk 2]
```diff
@@ -39,7 +39,29 @@ namespace nitro
...
+        if (groups_.find(group_name) == groups_.end())
+        {
+            groups_.emplace(group_name, option_group(group_name, description));
+        }
+        else if (description.size())
+        {
+            raise<parser_error>("Trying to redefine group. Name: ", group_name);
+        }
+
+        return group_assigner(group_name, *this);
```
Comment (bmario) [inline include/nitro/broken_options/parser.ipp:54]: With the suggested changes to the `option_group` and `group_assigner` classes, this should return the entry of the groups_ member.

**Comment by cvonelm** [comment]
Exactly what I've wanted, thank you! @bmario

**Pull request closed without merging**